### ISSUE TITLE
"Did not expect BoundT" error with function argument destructuring

### ISSUE THREAD

**Issue by fixplz** [facebook/flow#1347]
This:
```js
function f <Val> ({list} : {list : Array<Val>}) {
  return list[0]
}
```
fails with:
```
2.js:0
  0:
     ^ Failure("Did not expect BoundT")
```

Using Flow 0.20.1.

**Comment by samwgoldman** [comment]
Hey, you're in luck! This is fixed in the next release. Coming your way very soon now.

**Comment by fixplz** [comment]
Fixed in 0.21.0
